docs/contributing.md

```diff
@@ -0,0 +1,115 @@
+# Contributing to the monorepo
+
+## Getting started
+
+- Install [Node.js](https://nodejs.org) version 14.
+  - If you're using [NVM](https://github.com/creationix/nvm#installation) (recommended), `nvm use` will ensure that the right version is installed.
+- Install [Yarn v3](https://yarnpkg.com/getting-started/install).
+- Run `yarn install` to install dependencies and run any required post-install scripts.
+- Run `yarn simple-git-hooks` to add a [Git hook](https://github.com/toplenboren/simple-git-hooks#what-is-a-git-hook) to your local development environment which will ensure that all files pass linting before you push a branch.
+
+## Testing
+
+### Running unit tests
+
+- Run `yarn workspace <workspaceName> run test` to run all tests for a package.
+- Run `yarn workspace <workspaceName> run jest --no-coverage <file>` to run a test file within the context of a package.
+- Run `yarn test` to run tests for all packages.
+
+> **Note**  
+> `workspaceName` here is the `name` field within a package's `package.json`, e.g., `@metamask/announcement-controller`, not the directory where it is located, e.g., `packages/announcement-controller`.
+
+### Integrating with other projects
+
+When developing changes to packages within this repository that a different project depends upon, you may wish to load those changes into the project and test them locally or in CI before publishing proper releases of those packages. To solve that problem, this repository provides a mechanism to publish "preview" versions of packages to GitHub Package Registry. These versions can then be used in the project like any other version, provided the project is configured to use that registry.
+
+#### As a MetaMask contributor
+
+If you're a MetaMask engineer, you can create these preview versions via draft pull requests:
```

Comment:
Nit: should use the same term here presumably

```suggestion
If you're a MetaMask contributor, you can create these preview versions via draft pull requests:
```

Side-note, I've mostly become accustomed to it now, but I still hesitate to use the term "engineer" to describe developers because in Canada it's a protected term, only to be used by professional engineers. At least in theory. Plus there are non-developer contributors too.

Reply:
Ah - good point! I'd forgotten about that.